### PR TITLE
Remove Interval and Window from View, only support Cumulative views.

### PR DESCRIPTION
**Pull request by ramonza**
Implements: https://github.com/census-instrumentation/opencensus-specs/issues/47
This still needs discussion with other languages but there is rough consensus on removing intervals in their current form and they are broken (#183) in Go right now anyway so I'd like to move ahead with this.

### CODE REVIEW

File: exporter/stats/stackdriver/stackdriver.go [hunk 1]
```diff
@@ -451,16 +432,7 @@ func newLabelDescriptors(keys []tag.Key) []*labelpb.LabelDescriptor {
...
-	var w stats.Window
-
-	switch md.MetricKind {
-	case metricpb.MetricDescriptor_DELTA:
-		w = stats.Interval{}
-	case metricpb.MetricDescriptor_CUMULATIVE:
-		w = stats.Cumulative{}
-	}
-
+func equalAggWindowTagKeys(md *metricpb.MetricDescriptor, agg stats.Aggregation, keys []tag.Key) error {
```
Comment (odeke-em) [inline exporter/stats/stackdriver/stackdriver.go]: s/equalAggWindowTagKeys/equalAggTagKeys/g
as this PR removes window

File: plugin/grpc/grpcstats/client_metrics.go [hunk 2]
```diff
@@ -149,142 +125,6 @@ func defaultClientViews() {
 		RPCClientResponseCountView,
 	)
 	// TODO(jbd): Add roundtrip_latency, uncompressed_request_bytes, uncompressed_response_bytes, request_count, response_count.
-
-	RPCClientRoundTripLatencyMinuteView, _ = stats.NewView(
```
Comment (odeke-em) [inline plugin/grpc/grpcstats/client_metrics.go]: What's the rationale behind us getting rid of non-cumulative views?

File: stats/example_test.go [hunk 3]
```diff
@@ -39,11 +38,10 @@ func Example_view() {
 
 	view, err := stats.NewView(
 		"my.org/views/openconns",
-		"open connections distribution over one second time window",
+		"open connections",
 		nil,
 		m,
 		stats.DistributionAggregation([]float64{0, 1000, 2000}),
-		stats.Interval{Duration: time.Second},
```
Comment (odeke-em) [inline stats/example_test.go]: Yeah, I used this accidentally in the example :)

File: stats/view.go [hunk 4]
```diff
@@ -26,8 +26,8 @@ import (
 	"go.opencensus.io/tag"
 )
 
-// View allows users to filter and aggregate the recorded events
-// over a time window. Each view has to be registered to enable
+// View allows users to filter and aggregate the recorded events.
+// Each view has to be registered to enable
 // data retrieval. Use NewView to initiate new views.
```
Comment (odeke-em) [inline stats/view.go]: Perhaps we can backspace and move the `data retrieval` line to the end of the previous line.

**Comment by odeke-em** [comment]
Some merge conflicts or perhaps a rebase needed as the exporter structure was changed.

**Comment by songy23** [comment]
Could you add an issue about this on opencensus-specs as well?

File: plugin/grpc/grpcstats/client_metrics.go (hunk 2 above)
Comment (ramonza) [reply plugin/grpc/grpcstats/client_metrics.go]: https://github.com/census-instrumentation/opencensus-specs/issues/47

**Comment by ramonza** [comment]
https://github.com/census-instrumentation/opencensus-specs/issues/47

**Comment by ramonza** [comment]
@odeke-em updated and resolved conflicts, PTAL

File: exporter/stackdriver/stats.go [hunk 5]
```diff
@@ -237,28 +227,26 @@ func (e *statsExporter) makeReq(vds []*stats.ViewData, limit int) []*monitoringp
...
 	m := vd.View.Measure()
 	agg := vd.View.Aggregation()
-	window := vd.View.Window()
 	tagKeys := vd.View.TagKeys()
 	viewName := vd.View.Name()
 
 	if md, ok := e.createdViews[viewName]; ok {
-		// Check agg, window and keys.
-		return equalAggWindowTagKeys(md, agg, window, tagKeys)
+		return equalAggWindowTagKeys(md, agg, tagKeys)
```
Comment (odeke-em) [inline exporter/stackdriver/stats.go]: We still never changed this helper s/equalAggWindowTagKeys/equalAggTagKeys/g

File: exporter/stackdriver/stats_test.go [hunk 6]
```diff
@@ -803,20 +766,3 @@ func newTestCumViewData(v *stats.View, start, end time.Time, data1, data2 stats.
 	}
 }
 
-func newTestDistViewData(v *stats.View, start, end time.Time) *stats.ViewData {
```
Comment (odeke-em) [inline exporter/stackdriver/stats_test.go]: Am not sure why we removed this helper? Aren't we still using Distribution aggregations in the tests?

File: plugin/grpc/grpcstats/client_metrics.go [hunk 7]
```diff
@@ -42,24 +42,6 @@ var (
 	RPCClientResponseBytesView    *stats.View
 	RPCClientRequestCountView     *stats.View
 	RPCClientResponseCountView    *stats.View
-
-	RPCClientRoundTripLatencyMinuteView *stats.View
```
Comment (odeke-em) [inline plugin/grpc/grpcstats/client_metrics.go]: Hmm, let's also coordinate removing these views and interval windows from the Java client /cc @songy23 @sebright

File: plugin/grpc/grpcstats/client_metrics.go (hunk 2 above)
Comment (odeke-em) [reply plugin/grpc/grpcstats/client_metrics.go]: Thanks for the link @ramonza.

File: exporter/stackdriver/stats.go (hunk 5 above)
Comment (ramonza) [reply exporter/stackdriver/stats.go]: oops, changing

File: exporter/stackdriver/stats_test.go (hunk 6 above)
Comment (ramonza) [reply exporter/stackdriver/stats_test.go]: It was only used in one place and I removed that reference so I also removed the function. That place was basically asserting that we don't create SD time series for Interval windows, an assertion that no longer made sense after the current change. I agree that we should have a test for distribution, but it looks like we don't have this right now and I don't really have the context to write one as part of this PR. So I'm going to add back the function and just add a TODO to write a proper test for SD distribution.

File: plugin/grpc/grpcstats/client_metrics.go (hunk 7 above)
Comment (ramonza) [reply plugin/grpc/grpcstats/client_metrics.go]: I would prefer not to block this PR on other languages. Interval views are currently broken in Go anyway.

**Comment by ramonza** [comment]
updated PTAL

**Comment by ramonza** [comment]
@bogdandrutu I'd like to get this if you're ok with it?

File: plugin/grpc/grpcstats/client_metrics.go (hunk 7 above)
Comment (sebright) [reply plugin/grpc/grpcstats/client_metrics.go]: /cc @dinooliva

**Comment by ramonza** [comment]
Seems like this needs more discussion, let's hold off on merging for now.

**Comment by ramonza** [comment]
we have consensus on this, I will resolve conflicts and then it can be merged

**Comment by ramonza** [comment]
This has been merged separately into dev

**Pull request closed without merging**